lib/react_client.dart

```diff
@@ -181,6 +191,8 @@ class ReactDartComponentFactoryProxy2<TComponent extends Component2> extends Rea
   /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
   /// consumption by the [react] library internals.
   static JsMap generateExtendedJsProps(Map props) {
+    if (props == null) return newObject();
```

Comment:
Do we need this null-check here?

I would rather update any call-sites that might pass in null as opposed to opening up the possibility of something passing in null on accident, and silently failing instead of getting a null-error.